app/email.py

```diff
@@ -1,21 +1,16 @@
-from threading import Thread
-
-from flask import current_app, render_template
+import os
+from flask import render_template
 from flask.ext.mail import Message
 from . import mail
+from app import create_app
 
 
-def send_async_email(app, msg):
+def send_email(recipient, subject, template, **kwargs):
```

Comment:
@abhisuri97 I think Dubin and I talked about this exact point a while back. It's important that the **caller** is explicit about whether she wants the email to send synchronously or not. It can be very confusing (and lead to tricky bugs) when a function `doX` happens to be non-blocking without any indication to the caller.